src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt

```diff
@@ -1654,4 +1654,28 @@ class RsErrorAnnotatorTest : RsAnnotatorTestBase(RsErrorAnnotator::class.java) {
         trait Trait {}
         impl Drop for Trait {}
     """)
+
+    fun `test impl Drop and derive Copy E0184`() = checkErrors("""
+        #[lang = "drop"]
+        trait Drop {}
+        #[lang = "copy"]
+        trait Copy {}
+
+        #[derive(<error descr="Cannot implement both Copy and Drop [E0184]">Copy</error>)]
+        struct Foo;
+
+        impl <error descr="Cannot implement both Copy and Drop [E0184]">Drop</error> for Foo {}
+    """)
+
+    fun `test impl Drop and impl Copy E0184`() = checkErrors("""
+        #[lang = "drop"]
+        trait Drop {}
+        #[lang = "copy"]
+        trait Copy {}
+
+        struct Foo;
```

Comment:
Let's add a test for generic struct `struct Foo<T>(T);`. I don't sure how `implLookup.canSelect` behaves with unfolded generic types

Reply:
@vlad20012 Done